### PR TITLE
FIX avoid sql error (issue #26342)

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

File: htdocs/compta/paiement/list.php [hunk 1]
```diff
@@ -187,7 +187,7 @@
 } else {
 	// DISTINCT is to avoid duplicate when there is a link to sales representatives
 	$sql = "SELECT DISTINCT p.rowid, p.ref, p.datep, p.fk_bank, p.amount, p.statut, p.num_paiement";
-	$sql .= ", c.code as paiement_code";
```
Comment (eldy) [inline htdocs/compta/paiement/list.php:190]: I do not experience any trouble on v18 branch.

Comment (hregis) [reply htdocs/compta/paiement/list.php:190]: @eldy click here to order by type of payment 

Capture d’écran 2024-01-19 à 22 27 35

Comment (hregis) [reply htdocs/compta/paiement/list.php:190]: @eldy i have this error when i click to reorder by type of payment, with a Mysql 5.7.39 not MariaDB !!

```
Requête dernier accès en base en erreur: SELECT DISTINCT p.rowid, p.ref, p.datep, p.fk_bank, p.amount, p.statut, p.num_paiement, c.code as paiement_code, ba.rowid as bid, ba.ref as bref, ba.label as blabel, ba.number, ba.account_number as account_number, ba.fk_accountancy_journal as accountancy_journal, s.rowid as socid, s.nom as name, s.email FROM llx_paiement as p LEFT JOIN llx_c_paiement as c ON p.fk_paiement = c.id LEFT JOIN llx_bank as b ON p.fk_bank = b.rowid LEFT JOIN llx_bank_account as ba ON b.fk_account = ba.rowid LEFT JOIN llx_paiement_facture as pf ON p.rowid = pf.fk_paiement LEFT JOIN llx_facture as f ON pf.fk_facture = f.rowid LEFT JOIN llx_societe as s ON f.fk_soc = s.rowid WHERE p.entity IN (1) ORDER BY c.libelle DESC LIMIT 21
Code retour dernier accès en base en erreur: DB_ERROR_3065
Information sur le dernier accès en base en erreur: Expression #1 of ORDER BY clause is not in SELECT list, references column 'dolibarr18.c.libelle' which is not in SELECT list; this is incompatible with DISTINCT
```

Comment (hregis) [reply htdocs/compta/paiement/list.php:190]: @eldy my config of Mysql : 

for innodb_strict_mode	`ON`

for sql_mode : 
`ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION`

**Comment by eldy** [comment]
I see.
The problem is not in the missing field (there is no reason to have this field in the response since it is not used). The problem is in the use of the DISTINCT necessary because we use a join to make a filter on the sale representative instead of using a EXISTS. It has been fixed in v19. Can you fix by a backport of the fix of v19 ?

**Comment by hregis** [comment]
@eldy it's ok now, thank you

**Pull request merged**